### PR TITLE
fixes for iOS 7 and earlier

### PR DESCRIPTION
**Pull request by MrCapone**
1. `cocos2d::Director` `_invalid` variable must be set to `false` on first init to prevent unwanted return on `CCEAGLView-ios::layoutSubviews` when it called first time.
Improves pull request: https://github.com/cocos2d/cocos2d-x/pull/16394
Fixes issue: https://github.com/cocos2d/cocos2d-x/issues/16876
Also tested that case: https://github.com/cocos2d/cocos2d-x/issues/14276 to prevert regression.

2. Creation and setting the `GLView` should be done after creating the `RootViewController` and after `CCEAGLView` was added to view hierarchy. Otherwise, it can cause a problem when using the landscape orientation, because view will return wrong bounds size.
Improves pull request: https://github.com/cocos2d/cocos2d-x/pull/15792
Fixes issue: https://github.com/cocos2d/cocos2d-x/issues/16565

Tested on:
iPod 4th gen iOS 6.1.6
iPod 5th gen iOS 7.1
iPad mini 2th gen iOS 9.3.5
iPhone 6 iOS 10.1.1

### CODE REVIEW

**Comment by minggo** [comment]
will you please modify other templates too？

**Comment by MrCapone** [comment]
@minggo 
Should i just copy iOS templates from cpp to js and lua? Because iOS templates contain a lot of changes and fixes and lua/js templates are pretty outdated.

UPD
I did so, but I didn't touch copyrights.

File: templates/js-template-default/frameworks/runtime-src/proj.ios_mac/ios/AppController.mm [hunk 1]
```diff
@@ -140,5 +140,125 @@ - (void)dealloc {
 }
 
 
+@end#import "AppController.h"
```
Comment (minggo) [inline templates/js-template-default/frameworks/runtime-src/proj.ios_mac/ios/AppController.mm]: should remove old codes then add new codes?

**Comment by MrCapone** [comment]
@minggo 
I reset broken commit. Then push valid commits. Also i make some tests:

- create blank cpp project with new templates. Build passed, runs on iOS 10.1.
- create blank js project with new templates. Build passed, runs on iOS 10.1.
- create blank lua project with new templates. Build crash on `/lib/runtime/FileServer.cpp` but i applied this fix https://github.com/cocos2d/cocos2d-x/pull/16816 (for test build only) and build passed successfully, runs on iOS 10.1.

**Comment by minggo** [comment]
thanks

**Pull request merged**